internal/ignition/installmanifests.go

```diff
@@ -173,13 +174,13 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte)
 		system.NewLocalSystemInfo(),
 	)
 
-	release, err := g.installerCache.Get(g.installerReleaseImageOverride, g.releaseImageMirror,
-		g.cluster.PullSecret, ocRelease, g.cluster.OpenshiftVersion)
+	release, err := g.installerCache.Get(ctx, g.installerReleaseImageOverride, g.releaseImageMirror,
+		g.cluster.PullSecret, ocRelease, g.cluster.OpenshiftVersion, *g.cluster.ID)
 	if err != nil {
 		return errors.Wrap(err, "failed to get installer path")
 	}
 	//cleanup resources at the end
-	defer release.Release()
+	defer release.Release(ctx)
```

Comment:
I think naming Release a cleanup method in this context it's confusing (it was already named that way)
Also, I think sending events when successful within this method, and when not from within `installerCache.Get` it's also confusing.
Couldn't we defer `sendReleaseMetricsEvents` before calling `installerCache.Get`? Am I missing anything here?

Reply:
Btw if we do this, as clusterID is needed only for informational purposes, we could avoid changing the `installerCache.Get` signature

Reply:
To be fair I am not sure `Done` is much more descriptive than `Release`. I was thinking something like `release.Remove`, `release.Cleanup` or something that allows us to understand what's that for without having to look inside. I would definitely split it from the event sending though

Reply:
Cleanup it is then...